sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnSchemaInformation.java

```diff
@@ -141,7 +150,27 @@ DoFnSchemaInformation withUnboxPrimitiveParameter(
                     elementT))
             .build();
 
-    return toBuilder().setElementConverters(converters).build();
+    return toBuilder()
+        .setElementConverters(converters)
+        .setFieldAccessDescriptor(getFieldAccessDescriptor())
```

Comment:
This shouldn't be necessary - only need to specify changed fields

Reply:
Done.